app/models/user/authentication.rb

```diff
@@ -0,0 +1,29 @@
+class User::Authentication < ApplicationRecord
```

Comment:
I'd like to avoid names like `LoginToken` for models cause it tries to represent a key attribute or implementation, not the core essence, and usually only further complicates things down the road (like with `HcbCode`s, etc.). I think of authentication as the _process_ of attempting to identify an unknown user, and I tried to make that explicit with `#succeeded?`, but we can change it to `AuthenticationAttempt` if you feel like it'd be confusing otherwise.